lib/tokenizer/rainscriptTokenizerRules.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * The tokenizer rules, ordered by priority. All tokenizer rules go here.
+ *
+ * @type {{pattern: RegExp, type: any}[]}
+ * @property {RegExp} pattern a regular expression to be matched by a rule
+ * @property {*}      type    the type of the next token that will be returned
+ *                            when the pattern matches correctly
+ */
+export const rainscriptTokenizerRules = [
+  // quoted string literals
+  { pattern: /'(\\.|[^\\'])*'/, type: 'STR_1' },
+  { pattern: /"(\\.|[^\\"])*"/, type: 'STR_2' },
+  // event name
+  { pattern: /\[\[[^[\]]*\]\]/, type: 'EVENT' },
```

Comment:
According to your documentation of the syntax, the event's name doesn't allow newlines `\n`, square brackets `[]`, and slashes `/`.

Reply:
Ok, will add check for newlines `\n` and open square bracket `[` for the next commit. The regex already discounts closing square brackets `]`.

However, for slashes `/`, event _names_ indeed won't allow them but they're allowed if the event name is after `goto`:

```
goto [[ folder/EVENT ]]
```

Such cases should be checked by the parser instead of the tokenizer since a prior keyword become part of the grammar check, if that's okay.